<launch>
  <!-- This launchfile should bring up a node that broadcasts a ros image
       transport on /webcam/image_raw -->

  <node namespace="ros2gst" pkg="ros22gst" exec="ros22gst_node" output="screen">
    <param name="topic" value="/mavgscam0/camera/image_raw"/>
    <!--param name="gscam_config" value=" queue ! videoconvert ! autovideosink"/-->
    <!--param name="gscam_config" value=" queue ! videoconvert ! x264enc ! rtph264pay ! udpsink host=127.0.0.1 port=8554 sync=false async=false"/-->
    <param name="gscam_config" value=" queue ! videoconvert ! nvh264enc ! h264parse config-interval=1 ! rtph264pay pt=96 name=pay0"/>
    <!--param name="gscam_config" value=" queue ! videoconvert ! nvh264enc ! h264parse config-interval=1 ! rtph264pay pt=96 ! udpsink host=127.0.0.1 sync=false async=false"/-->
    <param name="fps" value="30"/>
    <param name="width" value="640"/>
    <param name="height" value="480"/>
    <param name="encoding" value="rgb8"/>
    <param name="rtspport" value="8910"/>
    <param name="rtspmount" value="/live0"/>
    <param name="udpport" value="8555"/>
  <!-- mavros server parameter -->
  
  </node>

</launch>
